src/middlewared/middlewared/plugins/acme_protocol_/authenticators/ovh.py

```diff
@@ -0,0 +1,51 @@
+import logging
+
+from lexicon.providers.ovh import ENDPOINTS
+from certbot_dns_ovh._internal.dns_ovh import _OVHLexiconClient
+
+from middlewared.schema import accepts, Dict, Str, ValidationErrors
+from middlewared.service import skip_arg
+
+from .base import Authenticator
+
+
+logger = logging.getLogger(__name__)
+OVH_ENDPOINTS = tuple(ENDPOINTS.keys())
+
+class OVHAuthenticator(Authenticator):
+
+    NAME = 'OVH'
+    PROPAGATION_DELAY = 60
+    SCHEMA = Dict(
+        'OVH',
+        Str('application_key', empty=False, null=False, title='OVH Application Key'),
+        Str('application_secret', empty=False, null=False, title='OVH Application Secret'),
+        Str('consumer_key', empty=False, null=False, title='OVH Consumer Key'),
+        Str('endpoint', empty=False, default='ovh-eu', title='OVH Endpoint', enum=OVH_ENDPOINTS),
+    )
+
+    def initialize_credentials(self):
+        self.application_key = self.attributes.get('application_key')
+        self.application_secret = self.attributes.get('application_secret')
+        self.consumer_key = self.attributes.get('consumer_key')
+        self.endpoint = self.attributes.get('endpoint')
+
+    @staticmethod
+    @accepts(SCHEMA)
+    @skip_arg(count=1)
+    async def validate_credentials(middleware, data):
+        verrors = ValidationErrors()
```

Comment:
You can just return data here or maybe even better would be to modify the base class and return data there by default, so route53 authenticator and ovh authenticator can benefit from those changes.

This is assuming you don't have any validation to do here and all above fields are required.

Reply:
I indeed do not have any validation to do here, after all, this method, at least on the cloudflare authenticator, is used to make sure that we either have a global API key, or a n API token to use, no API calls are made.

I could add some validation by trying to perform an API call to OVH, and see the HTTP response code, but it strays away from the behaviour of other authenticators.

Side note : I've noticed that certbot includes clients for other well-known registrars, maybe I can implement authenticators for them as well (like DigitalOcean, Linode, Google and others, that may help other people registered with these companies to set up their domain names)